internal/campaigns/run_steps.go

```diff
@@ -478,7 +436,7 @@ func (stepCtx *StepContext) ToFuncMap() template.FuncMap {
 // StepResult represents the result of a previously executed step.
 type StepResult struct {
 	// Files are the changes made to files by the step.
-	Files StepChanges
+	Files *StepChanges
```

Comment:
If we make this a pointer I think we should also lower-case `Files` to `files` so that users can't use it with the templating functionality. The original reason why I didn't make it a pointer was because those can lead to nil-pointer panics when playing around with templating, but after that we introduced the helper methods, so I think we don't need to to expose `Files` anymore, but best to check in the docs.

Reply:
Agreed, and the fact that changing it only affects this file and its associated test file probably reinforces that.